Source/Core/VideoCommon/VideoBackendBase.cpp

```diff
@@ -200,61 +198,59 @@ u16 VideoBackendBase::Video_GetBoundingBox(int index)
   return result;
 }
 
-// This function is called at static initialization, so we can't rely on s_default_backend being set
+static VideoBackendBase* GetDefaultVideoBackend()
+{
+  const auto& backends = VideoBackendBase::GetAvailableBackends();
+  if (backends.empty())
+    return nullptr;
+  return backends.front().get();
+}
+
 std::string VideoBackendBase::GetDefaultBackendName()
 {
-#ifdef HAS_OPENGL
-  return OGL::VideoBackend::NAME;
-#elif defined(_WIN32)
-  return DX11::VideoBackend::NAME;
-#else
-  return Vulkan::VideoBackend::NAME;
-#endif
+  auto* default_backend = GetDefaultVideoBackend();
+  return default_backend ? default_backend->GetName() : "";
 }
 
-void VideoBackendBase::PopulateList()
+const std::vector<std::unique_ptr<VideoBackendBase>>& VideoBackendBase::GetAvailableBackends()
 {
-  // OGL > D3D11 > D3D12 > Vulkan > SW > Null
+  static auto s_available_backends = [] {
```

Comment:
Why do you need the lambda?

Reply:
The lambda is just a convenient way to make sure the initialization code only runs once.

Reply:
Ah right, it's static, that makes sense.  Thanks!